src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice/GasPriceOracle.cs

```diff
@@ -100,11 +147,22 @@ private IEnumerable<UInt256> GetGasPricesFromRecentBlocks(IEnumerable<Block> blo
                 int txFromCurrentBlock = 0;
                 bool eip1559Enabled = SpecProvider.GetSpec(currentBlock.Number).IsEip1559Enabled;
                 UInt256 baseFee = currentBlock.BaseFeePerGas;
-                IEnumerable<UInt256> effectiveGasPrices = 
-                    currentBlockTransactions.Where(tx => tx.SenderAddress != currentBlock.Beneficiary)
+                IEnumerable<UInt256> effectiveGasPrices ;
+                if (!calculateMaxPriorityFee)
+                {
+                    effectiveGasPrices = currentBlockTransactions.Where(tx => tx.SenderAddress != currentBlock.Beneficiary)
                         .Select(tx => tx.CalculateEffectiveGasPrice(eip1559Enabled, baseFee))
                         .Where(g => g >= IgnoreUnder)
                         .OrderBy(g => g);
+                }
+                else
+                {
+                    effectiveGasPrices = currentBlockTransactions.Where(tx => tx.SenderAddress != currentBlock.Beneficiary)
+                        .Select(tx => tx.CalculateMaxPriorityFeePerGas(eip1559Enabled, baseFee))
```

Comment:
this select is the only part that we are differentiating. Can we simplify and instead of calculateMaxPriorityFee and big if, have this Func<> as a method parameter? Or even simpler a Func<> that is either CalculateEffectiveGasPrice or CalculateMaxPriorityFeePerGas